### DISCUSSION TITLE
Linux Install Error

### DISCUSSION THREAD

**Discussion by Skeynen** [opentrack/opentrack#1485] [Q&A]
Hello everyone,
i found an error during the installation on Pop!_OS 22.04. Before you ask: Yes I followed the Build Guide (https://github.com/opentrack/opentrack/wiki/Building-on-Linux#compiling-and-running-opentrack)
Guide:
-Install first for Debian, Ubuntu, etc;        on Pop!_OS its: sudo apt install...
  ...cmake
  ...git
  ...qttools5-dev
  ...qtbase5-private-dev
  ...libprocps-dev
  ...libopencv-dev
-run in a terminal:
1. git clone https://github.com/opentrack/opentrack
2. cd opentrack/
3. mkdir build
4. cd build
5. cmake ..
6. ccmake . (go down with 'PG DOWN' until you see SDK_WINE - then press 'ENTER' (set to 'ON'; press 'c' to reconfigure) 

<
To open the file I had to install "cmake-curses-gui" and rerun the command (sudo apt install cmake-curses-gui)
>

7. make

<
With SDK_WINE set to true I ran into this Error:
"[ 71%] Generating opentrack-wrapper-wine.exe.so
/bin/sh: 1: wineg++: not found
make[2]: *** [proto-wine/CMakeFiles/wine-wrapper.dir/build.make:75: proto-wine/opentrack-wrapper-wine.exe.so] Fehler 127
make[1]: *** [CMakeFiles/Makefile2:2219: proto-wine/CMakeFiles/wine-wrapper.dir/all] Fehler 2
make: *** [Makefile:136: all] Fehler 2" --> F
[RunMake.txt](https://github.com/opentrack/opentrack/files/8801089/RunMake.txt)
ull terminal output is attached
if you undo the change to SDK_WINE (back to off)it works 
but you only get as Input:
"Easy Tracker 1.1
FreePIE UDP reciver
Fusion
Linux Joystick input
PointTracker 1.1
S2Bot reciver
Test tracker
UDP over network"
and as Output:
"FlightGear
UDP over network"
>

8. make install
9. cd install
10. cd bin
11. ./opentrack

<
Note that if you run into a missing qplatformnativeinterface.h you can fix that by downloading the release:"opentrack 2.3.0 prerelease 4" at https://github.com/opentrack/opentrack/releases/tag/opentrack-2.3-rc49p4 an copying the missing flie form /qxt-mini into you opentrack folder. Than create a folder named qpa in your opentrack folder and copy the file into there as well. That should fix that
>

If you can provid the file (opentrack-wrapper-wine.exe.so) so its possible to work around the Error that would be awsome.
I suppose without the Error there are more input and output options that the listed ones?
Thanks for your great work :
[RunMake.txt](https://github.com/opentrack/opentrack/files/8801094/RunMake.txt)

**Comment by cofilippo** [comment]
I have the same problem.

I got one bit further by executing: ```sudo apt install wine64-tools gcc-multilib g++-multilib```. (Kubuntu22.04)
But now I run into a problem that I can't make the .so:

```
[ 71%] Generating opentrack-wrapper-wine.exe.so
/usr/bin/ld: cannot find -ladvapi32: No such file or directory
/usr/bin/ld: cannot find -luser32: No such file or directory
/usr/bin/ld: cannot find -lwinecrt0: No such file or directory
/usr/bin/ld: cannot find -lkernel32: No such file or directory
/usr/bin/ld: cannot find -lntdll: No such file or directory
collect2: error: ld returned 1 exit status
winegcc: /usr/bin/g++ failed
make[2]: *** [proto-wine/CMakeFiles/wine-wrapper.dir/build.make:75: proto-wine/opentrack-wrapper-wine.exe.so] Error 2
make[1]: *** [CMakeFiles/Makefile2:2275: proto-wine/CMakeFiles/wine-wrapper.dir/all] Error 2
make: *** [Makefile:136: all] Error 2

```

Not sure what to do next. Either the build instructions are way off for Debian/Ubuntu when using the wine option or my system is not installed correctly?

**Reply by sthalik** [reply]
What happens if you remove `-m32` in https://github.com/opentrack/opentrack/blob/master/proto-wine/CMakeLists.txt#L23?

**Reply by cofilippo** [reply]
removing the -m32 flag will make it compile. 

When I run it it seems to work, but I can't figure out what Steam application ID to use. I tried the one for DCS (223750), but no luck.
